plugins/rapid7_insightidr/komand_rapid7_insightidr/triggers/get_new_alerts/trigger.py

```diff
@@ -0,0 +1,118 @@
+import insightconnect_plugin_runtime
+import time
+from .schema import GetNewAlertsInput, GetNewAlertsOutput, Input, Output, Component
+
+# Custom imports below
+import datetime
+import json
+from insightconnect_plugin_runtime.helper import clean
+from insightconnect_plugin_runtime.exceptions import PluginException
+from komand_rapid7_insightidr.util.endpoints import Alerts
+from komand_rapid7_insightidr.util.resource_helper import ResourceHelper
+from komand_rapid7_insightidr.util.constants import TOTAL_SIZE
+
+
+class GetNewAlerts(insightconnect_plugin_runtime.Trigger):
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name="get_new_alerts",
+            description=Component.DESCRIPTION,
+            input=GetNewAlertsInput(),
+            output=GetNewAlertsOutput(),
+        )
+
+    def run(self, params={}):
+        # START INPUT BINDING - DO NOT REMOVE - ANY INPUTS BELOW WILL UPDATE WITH YOUR PLUGIN SPEC AFTER REGENERATION
+        input_frequency = params.get(Input.FREQUENCY, 15)
+        # END INPUT BINDING - DO NOT REMOVE
+        self.logger.info("Get Alerts: trigger started")
+
+        # Set initial set for storing initial alert_rrn values
+        initial_alerts = set()
+
+        # Flag to track the first execution
+        first_execution = True
+
+        while True:
+            start_time = datetime.datetime.utcnow() - datetime.timedelta(minutes=20)
+
+            search = clean(
+                {
+                    "start_time": start_time.strftime("%Y-%m-%dT%H:%M:%SZ"),
+                    "leql": params.get(Input.LEQL),
+                    "terms": params.get(Input.TERMS),
+                }
+            )
+            data = clean(
+                {
+                    "search": search,
+                    "field_ids": params.get(Input.FIELD_IDS),
+                    "aggregates": params.get(Input.AGGREGATES),
+                }
+            )
+
+            self.connection.session.headers["Accept-version"] = "strong-force-preview"
+            request = ResourceHelper(self.connection.session, self.logger)
+
+            endpoint = Alerts.get_alert_serach(self.connection.url)
+            response = request.resource_request(endpoint, "post", payload=data)
```

Comment:
Adding some error handling around the request is also advised, as a temporary 500 error that might resolve itself quickly should not cause the trigger to stop running